python/pyspark/rdd.py

```diff
@@ -293,21 +363,21 @@ def __getnewargs__(self):
         )
 
     @property
-    def context(self):
+    def context(self) -> "SparkContext":
         """
         The :class:`SparkContext` that this RDD was created on.
         """
         return self.ctx
 
-    def cache(self):
+    def cache(self: "RDD[T]") -> "RDD[T]":
```

Comment:
Covariant types cannot be reliably used, so I used `T` where arbitrary `RDD` is used.